.circleci/main.yml

```diff
@@ -1585,26 +1585,55 @@ jobs:
           command: |
             pip3 install -r requirements.txt
             PACHD_ADDRESS=$(minikube ip):30080 python3 -m pytest
-  bazel-tests:
+  bazel-style-tests:
     resource_class: medium
     docker:
       - image: debian:stable
         entrypoint: /bin/bash
+    working_directory: /code
+    steps:
+      - run: apt update
+      - run: DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true apt install -y wget git build-essential python3
+      - checkout
+      - run: |
+          pushd /tmp
+          wget https://github.com/bazelbuild/bazelisk/releases/download/v1.19.0/bazelisk-linux-amd64
+          chmod a+x bazelisk-linux-amd64
+          ln -s $PWD/bazelisk-linux-amd64 /usr/bin/bazel
+          useradd bazel -m
+          chown bazel -R /code
+          popd
+      - run: su bazel -c 'bazel run --remote_header=x-buildbuddy-api-key=${BUILDBUDDY_API_KEY} //:make_proto'
+      - run: su bazel -c 'git diff --exit-code'
+      - run: su bazel -c 'bazel run --remote_header=x-buildbuddy-api-key=${BUILDBUDDY_API_KEY} //:gazelle'
+      - run: su bazel -c 'git diff --exit-code'
+      - run: su bazel -c 'bazel test --remote_header=x-buildbuddy-api-key=${BUILDBUDDY_API_KEY} //:buildifier_test'
+      - run: cp -a bazel-testlogs/ /tmp/bazel-testlogs # store_test_results can't handle bazel-testlogs being a symlink
+      - store_test_results:
+          path: /tmp/bazel-testlogs/
+  bazel-tests:
+    resource_class: xlarge
+    docker:
+      - image: debian:stable
+        entrypoint: /bin/bash
+    working_directory: /code
     steps:
       - run: apt update
-      - run: DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true apt install -y wget git build-essential
+      - run: DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true apt install -y wget git build-essential python3
       - checkout
       - run: |
           pushd /tmp
           wget https://github.com/bazelbuild/bazelisk/releases/download/v1.19.0/bazelisk-linux-amd64
           chmod a+x bazelisk-linux-amd64
           ln -s $PWD/bazelisk-linux-amd64 /usr/bin/bazel
+          useradd bazel -m
+          chown bazel -R /code
           popd
-      - run: bazel run --remote_header=x-buildbuddy-api-key=${BUILDBUDDY_API_KEY} //:make_proto
-      - run: git diff --exit-code
-      - run: bazel test --remote_header=x-buildbuddy-api-key=${BUILDBUDDY_API_KEY} ...
+      - setup_remote_docker
+      - run: su bazel -c 'bazel test --remote_header=x-buildbuddy-api-key=${BUILDBUDDY_API_KEY} //oci:all'
```

Comment:
Some background... to build OCI images, you need tar.  `rules_pkg` implements this in python, so python is required.  python gets mad about running as root, so I changed everything to `su bazel -c` to run as a user.

Incidentally, we do version python in the MODULE.bazel, but `py_binary` targets have a wrapper written in python, which requires the host python to exist.  It's so bad.  When the "make a tar" thing is run, we use host python to find bazel's python, then exec using bazel's python.  I don't know what they were thinking.  (Similarly fun, the shell script wrapper is a C++ binary... which requires a C++ compiler to compile.  What?  Why not write both of those things in Java and build them into Bazel 😂 )

Reply:
😢